### ISSUE TITLE
Add interpolation support on showNotification()

### ISSUE THREAD

**Issue by LorenzHerz** [marmelab/admin-on-rest#1351]
Currently the Notification component just passes the translated message to the Snackbar.
```js
class Notification extends React.Component {
    handleRequestClose = () => {
        this.props.hideNotification();
    };

    render() {
        const style = {};
        const { primary1Color, accent1Color } = getStyles(this.context);
        const { type, translate, message } = this.props;
        if (type === 'warning') {
            style.backgroundColor = accent1Color;
        }
        if (type === 'confirm') {
            style.backgroundColor = primary1Color;
        }
        return (
            <Snackbar
                open={!!message}
                message={!!message && translate(message)}
                autoHideDuration={4000}
                onRequestClose={this.handleRequestClose}
                bodyStyle={style}
            />
        );
    }
}
```
However, if someone wants to interpolate a string into the Notification like so:
```js
showNotification('myroot.hello.world', {name: 'Planet Earth'});
```
It won't work, because the second argument of the `showNotificationAction` should be the type of notification. 

So I propose an additional optional parameter like so: 
```js
showNotification('myroot.hello.world', 'info', {name: 'Planet Earth'})
```

**Comment by fzaninotto** [comment]
Good idea!

**Comment by tkvw** [comment]
I've already added this in the last PR @see: https://github.com/marmelab/admin-on-rest/blob/d82fb2dc9df9d75c5922197a07fd49dfe977128d/packages/react-admin/src/mui/layout/Notification.js#L65
You can send `{message: 'myroot.hello.world',messageArgs:{name: 'Planet Earth'}}`
Maybe it should be documented

**Comment by fzaninotto** [comment]
Right, this was implemented in #1245. It was my mistake to merge it without doc. @tkvw, would you mind adding documentation about it?

**Comment by tkvw** [comment]
@LorenzHerz : actually you should use 
```
showNotification('myroot.hello.world', 'info',{messageArgs:{name: 'Planet Earth'}})
```
`autoHideDuration` is also supported in the options. 

@fzaninotto : where do you want the docs to be added? Currently there is no notification part in the docs.

**Comment by fzaninotto** [comment]
In the I18N docs

**Comment by fzaninotto** [comment]
Fixed by #1577
